src/NumberTheory/vinberg.jl

```diff
@@ -353,3 +353,163 @@ function vinberg_algorithm(S::ZZLat, upper_bound; v0=QQ[0;]::QQMatrix, root_leng
   return [r * B for r in roots]
 end
 
+
+function _get_tau(f::QQMatrix, Qb) ::QQBarFieldElem
```

Comment:
```suggestion
function _get_tau(f::QQMatrix, Qb)
```
Just some minor (technical) comment: There is no need to add those type annotations. They can in fact be "harmful" and thus we do not use them at all in the project.  Might be best to just remove them.

Reply:
To expand on Tommy's comment, this isn't even a "type annotation" but rather instructs Julia to perform a type conversion (via `convert`) on the return value, which in this case covers up the type instability of the code below. This can be fixed in a better by changing the initial value of `tau` (I'll annotate it there).